 1 ColorEngine: theme improvements/completeness
Use ThemeOverlay.Material.Light for light themes as parent: this fixes
incomplete theming when applying it to an app that has a dark theme by
default.
Also remove some of the colors and inherit them from the parent
instead, e.g. @color/white is not so good as text color for dark
themes since disabled preferences still kind of look enabled with
this, and now we have a nicer ripple color
 2 Partially revert "ColorEngine: theme improvements/completeness"
Using ThemeOverlay as parent for the light theme can cause issues when
applied to elements that are already dark in stock theme, so don't use
any parent here again.
Just make sure you're not theming dark applications without extra
measures.
Keep the dark theme colors improvement (inheritted from device default
(which is dark))

diff --git a/core/res/res/values/cm_themes.xml b/core/res/res/values/cm_themes.xml
index 002b6a74aed..62d24ecc56a 100644
--- a/core/res/res/values/cm_themes.xml
+++ b/core/res/res/values/cm_themes.xml
@@ -16,18 +16,15 @@
 -->
 <resources>
     <!-- begin Dark mode -->
-    <style name="Theme.DeviceDefault.Dark">
+    <style name="Theme.DeviceDefault.Dark" parent="Theme.DeviceDefault">
         <item name="alertDialogTheme">@style/Theme_Dark_Alert_Dialog</item>
 
         <!-- Color palette -->
         <item name="colorAccent">@color/accent_device_default_light</item>
         <item name="colorBackground">@color/dark_primary_color</item>
-        <item name="colorControlHighlight">@color/white</item>
-        <item name="colorControlNormal">@color/white</item>
         <item name="colorPrimary">@color/dark_primary_color</item>
         <item name="colorPrimaryDark">@color/dark_primary_color</item>
         <item name="colorSecondary">@color/dark_primary_color</item>
-        <item name="textColorPrimary">@color/white</item>
     </style>
 
     <!-- Need to use parent of original theme, otherwise attributes will be lost -->
@@ -41,18 +38,15 @@
     <!-- End dark mode -->
 
     <!-- begin Grey mode -->
-    <style name="Theme.DeviceDefault.Grey">
+    <style name="Theme.DeviceDefault.Grey" parent="Theme.DeviceDefault">
         <item name="alertDialogTheme">@style/Theme_Grey_Alert_Dialog</item>
 
         <!-- Color palette -->
         <item name="colorAccent">@color/accent_device_default_light</item>
         <item name="colorBackground">@color/grey_primary_color</item>
-        <item name="colorControlHighlight">@color/white</item>
-        <item name="colorControlNormal">@color/white</item>
         <item name="colorPrimary">@color/grey_primary_color</item>
         <item name="colorPrimaryDark">@color/grey_primary_color</item>
         <item name="colorSecondary">@color/grey_primary_color</item>
-        <item name="textColorPrimary">@color/white</item>
     </style>
 
     <!-- Need to use parent of original theme, otherwise attributes will be lost -->
@@ -66,18 +60,15 @@
     <!-- End grey mode -->
 
     <!-- Begin white mode -->
-    <style name="Theme_DeviceDefault_White">
+    <style name="Theme_DeviceDefault_White" parent="ThemeOverlay.Material.Light">
         <item name="alertDialogTheme">@style/Theme_White_Alert_Dialog</item>
 
         <!-- Color palette -->
         <item name="colorAccent">@color/accent_device_default_light</item>
         <item name="colorBackground">@color/background_material_light</item>
-        <item name="colorControlHighlight">@color/ripple_material_light</item>
-        <item name="colorControlNormal">@color/secondary_text_material_light</item>
         <item name="colorPrimary">@color/primary_device_default_settings</item>
         <item name="colorPrimaryDark">@color/primary_dark_device_default_settings</item>
         <item name="colorSecondary">@color/secondary_device_default_settings</item>
-        <item name="textColorPrimary">@color/primary_text_material_light</item>
     </style>
 
     <style name="Theme_White_Alert_Dialog" parent="Theme.DeviceDefault.Light.Dialog.Alert">
diff --git a/core/res/res/values/cm_themes.xml b/core/res/res/values/cm_themes.xml
index 012b6a74aed..62d24ecc56a 100644
--- a/core/res/res/values/cm_themes.xml
+++ b/core/res/res/values/cm_themes.xml
@@ -16,7 +16,7 @@
 -->
 <resources>
     <!-- begin Dark mode -->
-    <style name="Theme.DeviceDefault.Dark" parent="Theme.DeviceDefault">
+    <style name="Theme.DeviceDefault.Dark">
         <item name="alertDialogTheme">@style/Theme_Dark_Alert_Dialog</item>
 
         <!-- Color palette -->
@@ -38,7 +38,7 @@
     <!-- End dark mode -->
 
     <!-- begin Grey mode -->
-    <style name="Theme.DeviceDefault.Grey" parent="Theme.DeviceDefault">
+    <style name="Theme.DeviceDefault.Grey">
         <item name="alertDialogTheme">@style/Theme_Grey_Alert_Dialog</item>
 
         <!-- Color palette -->
@@ -60,15 +60,18 @@
     <!-- End grey mode -->
 
     <!-- Begin white mode -->
-    <style name="Theme_DeviceDefault_White" parent="ThemeOverlay.Material.Light">
+    <style name="Theme_DeviceDefault_White">
         <item name="alertDialogTheme">@style/Theme_White_Alert_Dialog</item>
 
         <!-- Color palette -->
         <item name="colorAccent">@color/accent_device_default_light</item>
         <item name="colorBackground">@color/background_material_light</item>
+        <item name="colorControlHighlight">@color/ripple_material_light</item>
+        <item name="colorControlNormal">@color/secondary_text_material_light</item>
         <item name="colorPrimary">@color/primary_device_default_settings</item>
         <item name="colorPrimaryDark">@color/primary_dark_device_default_settings</item>
         <item name="colorSecondary">@color/secondary_device_default_settings</item>
+        <item name="textColorPrimary">@color/primary_text_material_light</item>
     </style>
 
     <style name="Theme_White_Alert_Dialog" parent="Theme.DeviceDefault.Light.Dialog.Alert">
